libbeat/processors/add_host_metadata/add_host_metadata.go

```diff
@@ -71,10 +84,68 @@ func (p *addHostMetadata) loadData() {
 		if p.info.OS.Build != "" {
 			p.data.Put("host.os.build", p.info.OS.Build)
 		}
+
+		if p.config.NetInfoEnabled {
+			// IP-address and MAC-address
+			var ipList, hwList, err = p.getNetInfo()
+			if err != nil {
+				logp.Warn("Error when getting network information %v", err)
+			}
+
+			if len(ipList) > 0 {
+				p.data.Put("host.ip", ipList)
+			}
+			if len(hwList) > 0 {
+				p.data.Put("host.mac", hwList)
+			}
+		}
+
 		p.lastUpdate = time.Now()
 	}
 }
 
+func (p addHostMetadata) getNetInfo() ([]string, []string, error) {
+	var ipList []string
+	var hwList []string
+
+	// Get all interfaces and loop through them
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return ipList, hwList, err
```

Comment:
Can you use `return nil, nil, err` here? This makes it clear that both ipList and hwList do not have any values yet. Code logic is exactly the same as before.